src/main/java/com/google/sps/data/RentPost.java

```diff
@@ -0,0 +1,152 @@
+package com.google.sps;
+
+import static com.google.sps.Constants.CURRENCY_CODE;
+import static com.google.sps.Constants.DATE_FORMAT;
+import static com.google.sps.Constants.RENT_COLLECTION_NAME;
+import static com.google.sps.Constants.REQUEST_DESCRIPTION;
+import static com.google.sps.Constants.REQUEST_END_DATE;
+import static com.google.sps.Constants.REQUEST_LEASE_TYPE;
+import static com.google.sps.Constants.REQUEST_NUM_ROOMS;
+import static com.google.sps.Constants.REQUEST_PRICE;
+import static com.google.sps.Constants.REQUEST_ROOM_TYPE;
+import static com.google.sps.Constants.REQUEST_START_DATE;
+import static com.google.sps.Constants.REQUEST_TIMESTAMP;
+import static com.google.sps.Constants.REQUEST_TITLE;
+
+import com.google.auto.value.AutoValue;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.FieldValue;
+import com.google.cloud.Timestamp;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+import javax.money.CurrencyUnit;
+import javax.money.Monetary;
+import org.javamoney.moneta.Money;
+
+/** A rent post made by a user. */
+@AutoValue
+public abstract class RentPost {
```

Comment:
nit: RentPost is an odd name - that's where it comes from, but not really what it is. Listing, maybe? HousingListing?

Reply:
Changed class name to listing and all other rent references to listing.